sql/executor.go

```diff
@@ -429,6 +439,56 @@ func (p parameters) Arg(name string) (parser.Datum, bool) {
 	}
 }
 
+var _ parser.Args = golangParameters{}
+
+type golangParameters []interface{}
```

Comment:
OK. I dislike the type switch whack-a-mole below, but it's probably the best you can do if you're against pushing the compiler appeasement to the caller.

LGTM